From c9dc15fd5e8ea4d562a48dd9180f952d555d04c9 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 02 Oct 2019 10:47:19 +0300
Subject: [PATCH] Revert "[SW-0] version: move string to SRAM"

Since we are going to move back to SRAM, the automatic allocation is
better than the hard coded offset for the version string.
Revert the commit to get back to the original implementation.

This reverts commit c558fda8dbd6d5eff5244df6740cc850c2354e77.

Change-Id: I1dfe6b5c9e8c0ce6514db7b09c9d1f33901874ac
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/preboot.c b/app/preboot.c
index 2abfb1a..d51d909 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -97,39 +97,36 @@
 }
 
 #ifndef TARGET_IS_TESTER
+static const char *preboot_ver_str = PREBOOT_VER_STR;
+static char preboot_btl_ver_str[VERSION_MAX_LEN] = "BTL version ";
+
 static void preboot_version(void)
 {
-	const char *btl_ver_tag = "BTL version ";
-	const char *preboot_ver_str = PREBOOT_VER_STR;
-	char *dst_version_str = (char *)PREBOOT_VER_BASE;
-	unsigned long offset = (unsigned long)dst_version_str;
+	u64 offset = (u64) preboot_btl_ver_str;
 	u32 btlver = regmap_readl(mmPSOC_GLOBAL_CONF_SCRATCHPAD_28);
 	int i, btl_ver_sz;
 
 	preboot_puts("\n");
-	preboot_puts_raw(btl_ver_tag);
+	preboot_puts_raw(preboot_btl_ver_str);
 	preboot_puth(btlver, true);
 	preboot_puts_raw(preboot_ver_str);
 	preboot_puts("\n");
 
-	/* Copy the BTL version tag to destination (SRAM) address */
-	for (i = 0; i < VERSION_MAX_LEN; i++) {
-		if (btl_ver_tag[i] == '\0')
+	/* get to the end of the predefined btl string */
+	for (i = 0; i < VERSION_MAX_LEN; i++)
+		if (preboot_btl_ver_str[i] == '\0')
 			break;
 
-		dst_version_str[i] = btl_ver_tag[i];
-	}
-
 	/* convert and copy the btl version */
-	i += preboot_uint2hexstring(&dst_version_str[i], btlver, true);
+	i += preboot_uint2hexstring(&preboot_btl_ver_str[i], btlver, true);
 
 	/* separate the btl from preboot */
-	dst_version_str[i++] = '\n';
+	preboot_btl_ver_str[i++] = '\n';
 	btl_ver_sz = i;
 
 	/* copy the preboot version */
 	for (int j = 0; j < VERSION_MAX_LEN - btl_ver_sz - 1; j++) {
-		dst_version_str[i++] = preboot_ver_str[j];
+		preboot_btl_ver_str[i++] = preboot_ver_str[j];
 		if (preboot_ver_str[j] == '\0')
 			break;
 	}
